config/pay-translation.js

```diff
@@ -0,0 +1,21 @@
+'use strict'
+// i18n’s built in translation function `__()` returns the key
+// if it can’t find a translation. This is problematic as we show
+// something gross like 'cardDetails.title' to the user.
+// So this is a custom function that will check that the string
+// doesn’t equal the key. And if it does throw an error so the
+// template fails to render but with a nice helpful error message.
+
+// NPM dependencies
+const i18n = require('i18n')
+
+module.exports = function (req, res, next) {
+  res.locals.__p = res.__p = function () {
+    const translation = i18n.__.apply(req, arguments)
+    if (arguments[0] === translation) {
+      throw new Error(`Template is missing a translation with ID: ${translation}`)
```

Comment:
Depending on the behaviour we need - ideally we should not be throwing this error from middleware. The `next` middleware model designed to handle errors that occur during the express stack. 

This would look something like: 
```js 
const keyNotTranslated = arguments[0] === translation

if (keyNotTranslated) { 
  // no further middleware or controllers will be visited - this ends the flow
  next(new Error(`Template is missing a translation with ID: ${transaction}`))
  return
}

// everything is okay - continue processing the express stack
next()
```

Reply:
I am not sure I understand this? This is part of the middleware stack?